src/js/views/CovidDashboardView.js

```diff
@@ -0,0 +1,881 @@
+      const props = [];
+      this.properties.forEach((property) => {
+        let active = '';
+        if (property.name === this.properties[0].name) active = 'cell-active';
+        const prop = elementFactory('div',
+          { class: `table-cell cell-numeric ${active}`, 'data-property': property.name },
+          `${country[property.name]}`);
+        props.push(prop);
+      });
+      const flag = elementFactory('img', { src: country.flag, class: 'flag-img' }, '');
+      const row = elementFactory('div', { class: 'table-row' }, flag, name, ...props);
+
+      row.onclick = () => {
+        // eslint-disable-next-line no-alert
+        // alert(`${country.Country} || ${country.CountryCode}`);
+        this.moveTheMap(country.CountryCode);
+        this.selectedCountry = country.CountryCode;
+        this.updateCovidInfoTable();
+        this.emit('updatedata', country.Country);
+      };
+
+      rows.push(row);
+    });
+    rows.forEach((row) => {
+      this.table.appendChild(row);
+    });
+    this.sortTable();
+    this.properties = properties.filter((prop) => prop.isLastDay === this.isLastDay && prop.isPerPopulation === this.isPopulation);
+  }
+
+  updateChart() {
+    if (!this.chartData[0]) {
+      return;
+    }
+    this.aroundTheWorldCases.Cases_of_Infection = [];
+    this.aroundTheWorldCases.Cases_of_Deaths = [];
+    this.aroundTheWorldCases.Cases_of_Recovery = [];
+    if (this.selectedCountry) {
+      this.aroundTheWorldCases.Dates_of_Updating = [];
+      const countryInfo = this.model.data.CountriesInfo.find((item) => item.CountryCode === this.selectedCountry);
+      const countryPopulationPer100k = countryInfo.population / 100000;
+      this.chartData.forEach((item) => {
+        if (this.isPopulation) {
+          this.aroundTheWorldCases.Cases_of_Infection.push((item.Confirmed / countryPopulationPer100k).toFixed(5));
+          this.aroundTheWorldCases.Cases_of_Deaths.push((item.Deaths / countryPopulationPer100k).toFixed(5));
+          this.aroundTheWorldCases.Cases_of_Recovery.push((item.Recovered / countryPopulationPer100k).toFixed(5));
+        } else {
+          this.aroundTheWorldCases.Cases_of_Infection.push(item.Confirmed);
+          this.aroundTheWorldCases.Cases_of_Deaths.push(item.Deaths);
+          this.aroundTheWorldCases.Cases_of_Recovery.push(item.Recovered);
+        }
+        this.aroundTheWorldCases.Dates_of_Updating.push(`${item.Date.split('-')[1]} ${item.Date.split('-')[0]}`);
+      });
+      this.myChart.data.labels = this.aroundTheWorldCases.Dates_of_Updating;
+      this.myChart.options.title.text = `${this.chartData[0].Country}`;
+    } else {
+      const earthPopulationPer100k = this.model.data.GlobalInfo.earth_population / 100000;
+      this.chartData.forEach((item) => {
+        if (this.isPopulation) {
+          this.aroundTheWorldCases.Cases_of_Infection.push((item.total_cases / earthPopulationPer100k).toFixed(5));
```

Comment:
i think i've seen this already and this could be probably moved to a helper function and reused for avoiding code duplication